test/workbox-core/sw/_private/test-cacheWrapper.mjs

```diff
@@ -366,5 +443,78 @@ describe(`cacheWrapper`, function() {
       expect(spyOne.callCount).to.equal(1);
       expect(spyTwo.callCount).to.equal(1);
     });
+
+    it(`should call cacheKeyWillBeUsed`, async function() {
```

Comment:
Repeat code? Looks like these two tests are the same as the ones above.

Reply:
They're in different `describe()` blocks—one for `.put()` and the other for `.match()`. The actual test cases are different, too.

Reply:
In particular, the tests confirm that the correct `mode` value (`'read'` or `'write'`) is passed in to the plugin, and that the underlying `cache.match()` or `cache.put()` operation is actually called with the plugin's return value.

Reply:
Ahh, I see. I couldn't easily see the outer blocks from the diff.